File: app/src/main/java/mb00/android/codehub/ui/gist/adapter/GistFragmentPagerAdapter.java
package mb00.android.codehub.ui.gist.adapter;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import mb00.android.codehub.R;
import mb00.android.codehub.ui.gist.view.GistActivity;
import mb00.android.codehub.ui.gist.view.GistCommentsFragment;
import mb00.android.codehub.ui.gist.view.GistFilesFragment;

/**
 * FragmentPagerAdapter used to display gist-related Fragments; launched from {@link GistActivity}
 */

public class GistFragmentPagerAdapter extends FragmentPagerAdapter {

    //==============================================================================================
    // GistFragmentPagerAdapter fields
    //==============================================================================================

    private final int PAGE_COUNT = 2;

    private String filesTabTitle;
    private String commentsTabTitle;

    private Bundle gistArgs;

    //==============================================================================================
    // GistFileAdapter constructor
    //==============================================================================================

    public GistFragmentPagerAdapter(FragmentManager fragmentManager, Context context, Bundle gistBundle) {
        super(fragmentManager);

        filesTabTitle = context.getResources().getString(R.string.files);
        commentsTabTitle = context.getResources().getString(R.string.comments);

        gistArgs = gistBundle;
    }

    //==============================================================================================
    // FragmentPagerAdapter methods
    //==============================================================================================

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        switch(position) {
            case 0:
                GistFilesFragment filesFragment = new GistFilesFragment();
                filesFragment.setArguments(gistArgs);
                return filesFragment;
            case 1:
                GistCommentsFragment commentsFragment = new GistCommentsFragment();
                commentsFragment.setArguments(gistArgs);
                return commentsFragment;
            default:
                return null;
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch(position) {
            case 0:
                return filesTabTitle;
            case 1:
                return commentsTabTitle;
            default:
                return null;
        }
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

}

File: app/src/main/java/mb00/android/codehub/ui/gist/adapter/GistFragmentPagerAdapter.kt
package mb00.android.codehub.ui.gist.adapter

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import mb00.android.codehub.R
import mb00.android.codehub.ui.gist.view.GistActivity
import mb00.android.codehub.ui.gist.view.GistCommentsFragment
import mb00.android.codehub.ui.gist.view.GistFilesFragment

/**
 * FragmentPagerAdapter used to display gist-related Fragments; launched from [GistActivity]
 */

class GistFragmentPagerAdapter(
        private val fragmentManager: FragmentManager,
        private val context: Context,
        private val gistArgs: Bundle?
) : FragmentPagerAdapter(fragmentManager) {

    private val PAGE_COUNT = 2

    private val filesTabTitle: String = context.resources.getString(R.string.files)
    private val commentsTabTitle: String = context.resources.getString(R.string.comments)

    override fun getCount(): Int {
        return PAGE_COUNT
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> getGistFilesFragment()
            1 -> getGistCommentsFragment()
            else -> getGistFilesFragment()
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> filesTabTitle
            1 -> commentsTabTitle
            else -> null
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        return POSITION_NONE
    }

    private fun getGistFilesFragment(): GistFilesFragment {
        val filesFragment = GistFilesFragment()
        filesFragment.arguments = gistArgs
        return filesFragment
    }

    private fun getGistCommentsFragment(): GistCommentsFragment {
        val commentsFragment = GistCommentsFragment()
        commentsFragment.arguments = gistArgs
        return commentsFragment
    }

}
